openedx/core/djangoapps/credit/migrations/0008_auto__del_index_creditprovider_provider_id__chg_field_crediteligibilit.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+        # Changing field 'CreditEligibility.provider'
+        db.alter_column('credit_crediteligibility', 'provider_id', self.gf('django.db.models.fields.related.ForeignKey')(null=True, to=orm['credit.CreditProvider']))
+
+    def backwards(self, orm):
+        # Changing field 'CreditEligibility.provider'
+        db.alter_column('credit_crediteligibility', 'provider_id', self.gf('django.db.models.fields.related.ForeignKey')(default=1, to=orm['credit.CreditProvider']))
```

Comment:
Ah, okay, I missed that we were setting the default to 1 on this line.  I still think this is okay because:

1) This is a backwards migration, and we're not going to turn this feature on until the forward migration is successful (the `provider_id` field is set to NULL and then deleted).  This won't set the field to 1 once the forward migration completes, because the field won't exist :)

2) The open source community is not using this feature.  It hasn't been publicly release, and even if it had been, it's unclear that credit courses make sense in open source installations.